xbmc/network/NetworkServices.cpp

```diff
@@ -854,6 +853,42 @@ bool CNetworkServices::StopUPnPClient()
   return false;
 }
 
+bool CNetworkServices::StartUPnPController()
+{
+#ifdef HAS_UPNP
+  if (!CSettings::Get().GetBool("services.upnpcontroller") ||
+      !CSettings::Get().GetBool("services.upnpserver"))
```

Comment:
No because you could enable the `services.upnpcontroller` but then disable the `services.upnpserver` (which is the parent setting of `services.upnpcontroller`) so `services.upnpcontroller` would still be true but `services.upnpserver` would be false and then the controller would not work.